paasta_tools/cli/cmds/spark_run.py

```diff
@@ -1059,7 +1068,12 @@ def build_and_push_docker_image(args: argparse.Namespace) -> Optional[str]:
     if cook_return != 0:
         return None
 
-    docker_url = f"{args.docker_registry}/{docker_tag}"
+    registry_uri = args.docker_registry or _get_adhoc_docker_registry(
+        service=args.service,
+        soa_dir=args.yelpsoa_config_root,
+    )
```

Comment:
the intent here is to allow --docker-registry to override whatever is in yelpsoa